### ISSUE TITLE
Allow jlink from toolchain for create-runtime-image

### ISSUE THREAD

**Issue by jftsunami** [moditect/moditect#78]
It would be useful to allow the create-runtime-image goal to pickup the jlink executable from a toolchain like the existing <baseJdk> option does.  The goal currently fails when trying to jlink a Java 10 project while running Maven under Java 11, as the jlink spawned for the task is coming from the 11 JDK.

**Comment by gunnarmorling** [comment]
Yeah, that's a great idea. Would you be willing to help out by sending a pull request for this?

**Comment by jftsunami** [comment]
If I can get the bandwidth to implement the change.

**Comment by gunnarmorling** [comment]
Cool; just let me know how it goes or whether you need some help to get started.

**Comment by heatzync** [comment]
What this means is that the `moditect-maven-plugin` may potentially have to select 2 JDKs from the toolchain...

For example, if the `moditect-maven-plugin` is executed on Linux but it should build a v17 JRE for Windows, then it'll need to:
1. select the Linux v17 toolchain whose `jlink` will be used; and
2. select the Windows v17 toolchain whose `jmods` will be used.
